CMakeLists.txt

```diff
@@ -40,6 +40,11 @@ if(LIBPRESSIO_TOOLS_HAS_ERROR_INJECTOR)
   find_package(LibPressioErrorInjector REQUIRED)
   target_link_libraries(utils PUBLIC LibPressioErrorInjector::libpressio_error_injector)
 endif()
+option(LIBPRESSIO_TOOLS_HAS_MPI "require and link to the MPI module" OFF)
+if(LIBPRESSIO_TOOLS_HAS_MPI)
+	find_package(MPI REQUIRED COMPONENTS CXX)
+	target_link_libraries(pressio-tools PRIVATE MPI::MPI_CXX utils)
```

Comment:
I'm would be surprised  if this compiled if `-DLIBPRESSIO_TOOLS_HAVE_MPI=ON` was passed to `cmake`.

You don't need line 46 here, just line 45.  You add the link libraries in src/pressio/CMakeLists.txt